src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ISymbolExtensions.cs

```diff
@@ -364,13 +364,33 @@ public static bool IsParams([NotNullWhen(returnValue: true)] this ISymbol? symbo
         }
 
         public static ImmutableArray<IParameterSymbol> GetParameters(this ISymbol? symbol)
-            => symbol switch
+        {
+            return symbol switch
             {
                 IMethodSymbol m => m.Parameters,
-                IPropertySymbol nt => nt.Parameters,
+                IPropertySymbol p => p.Parameters,
+                INamedTypeSymbol nt when nt.IsRecord => GetRecordParameters(nt),
                 _ => ImmutableArray<IParameterSymbol>.Empty,
             };
 
+            ImmutableArray<IParameterSymbol> GetRecordParameters(INamedTypeSymbol recordSymbol)
+            {
+                const int RecordDeclarationRawKind = 9063;
+
+                // A bit hacky to determine the parameters of primary constructor associated with a given record.
+                // TODO: record structs.
+                var primaryConstructor = recordSymbol.InstanceConstructors.FirstOrDefault(
+                    c => c.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax()?.RawKind == RecordDeclarationRawKind);
+
+                if (primaryConstructor is null)
+                {
+                    return ImmutableArray<IParameterSymbol>.Empty;
+                }
+
+                return primaryConstructor.Parameters;
```

Comment:
Having primary constructor with `AssociatedSymbol` equals to the record symbol would save this ugly workaround :smile:

@jcouv @jaredpar Is this something you want to do in the compiler side?